### PR TITLE
Finish up localization

### PR DESCRIPTION
**Pull request by bellisk**

### CODE REVIEW

File: ckanext/switzerland/helpers/backend_helpers.py [hunk 1]
```diff
@@ -190,3 +183,58 @@ def ogdch_resource_display_name(res):
...
+    title = get_localized_value_for_display(title)
+    return _link_to(title, url)
+
+
+def resource_link(resource_dict, package_id):
+    """
+    monkey patched version of ckan.lib.helpers.resource_link which extracts the
+    correct translation of the resource name
+    """
+    log.warning('resource_link')
```
Comment (stefina) [inline ckanext/switzerland/helpers/backend_helpers.py:230]: Do we need the logging here? If so, I would like it if you could add information so that this is more useful to us.
Considering this might be called a lot, I assume it is a relict of debugging? ;)

File: ckanext/switzerland/templates/admin/trash.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{% ckan_extends %}
+
+{% block primary_content_inner %}
+  {% set truncate = truncate or 180 %}
+  {% set truncate_title = truncate_title or 80 %}
```
Comment (stefina) [inline ckanext/switzerland/templates/admin/trash.html:5]: I have never seen that notation, can you explain to me what setting it to 180 or 80 does here?

File: ckanext/switzerland/helpers/backend_helpers.py [hunk 3]
```diff
@@ -190,3 +183,58 @@ def ogdch_resource_display_name(res):
...
+    monkey patched version of ckan.lib.helpers.group_link which extracts the
+    correct translation of the group title
+    """
+    url = url_for(controller='group', action='read', id=group['name'])
+    title = group['title']
+    try:
+        # the group creation message contains str(dict), so we must parse the
+        # string to fix it. If the title is just a string, a ValueError is
+        # thrown.
+        title = ast.literal_eval(title)
```
Comment (stefina) [inline ckanext/switzerland/helpers/backend_helpers.py:218]: is `title` here a dict as we have to do the `get_localized_value_for_display` action anyways? 
maybe we can name it that, to make it more clear.

File: ckanext/switzerland/helpers/backend_helpers.py (hunk 1 above)
Comment (bellisk) [reply ckanext/switzerland/helpers/backend_helpers.py:230]: Oops, this is indeed a relict of debugging. I will clean this up!

File: ckanext/switzerland/templates/admin/trash.html (hunk 2 above)
Comment (bellisk) [reply ckanext/switzerland/templates/admin/trash.html:5]: This is copied over from the CKAN template - I think it's to make the template more flexible. You can either pass in values for `truncate` and `truncate_title`, or use the defaults (180 and 80). In line 9, `truncate_title` is used as the length to which the title should be truncated.

Looking at it now, it seems that we never use the variable `truncate`, and so far as I know we never use any truncate-length besides the default anyway. This could be cleaned up if you think it's confusing like this.

Comment (stefina) [reply ckanext/switzerland/templates/admin/trash.html:5]: ah, thank you for the explanation :) I think using the reasoning that we override from CKAN where we need it, I would leave it like that. Unless we need it differently, of course, but probably makes it easier to read commit-wise ('copy template'-commit and then everything we need differently)

TLDR: Let's keep it, thank you for explaining this to me 😄

File: ckanext/switzerland/helpers/backend_helpers.py (hunk 3 above)
Comment (bellisk) [reply ckanext/switzerland/helpers/backend_helpers.py:218]: Hm, I tried to leave an explanatory comment here but I don't think I succeeded! It seems that, at this point, `title` can either be a string (e.g. `'Administration'`) or a dictionary represented by a string (e.g. `'{"de": "Administration", "fr": "Administration but in French"}'`). The latter case only happens when we are handling the group creation message, i.e. what is shown on the Activity Stream page, but not when handling group edit or delete messages.

If we get the dictionary representation, we have to do `ast.literal_eval` to transform it back to a dict before localizing. If it's a string, we try to eval it, but get a ValueError, which we ignore. Passing a string to `get_localized_value_for_display` does eventually return the same string, so it's okay to do that, but doesn't seem very efficient. I guess we can move that call into the try block as well. :thinking:

I will try to improve the comment with some of what I wrote here, if that's okay.

Comment (stefina) [reply ckanext/switzerland/helpers/backend_helpers.py:218]: I think I understood that part, that we have to do that guessing anyways. 
But I think it would just make it a little more clear to state, that this is not the final title, but rather a `title_thing`, `raw_title` (something that indicates in the name, that there is a title somewhere, but we don't know yet, in which form) It is very nitpicking, though, if you have an idea, go for it, but if not, I am fine with leaving it like that, too 🙂

Comment (bellisk) [reply ckanext/switzerland/helpers/backend_helpers.py:218]: Thanks for explaining what you meant! I decided to keep the name `title` because it gets a bit complicated otherwise, with the many processing steps and the try/except. I hope this is all right for you.

I also added in the step here to decode accent characters, if necessary - of course this is only needed sometimes and not all the time :roll_eyes:

Comment (stefina) [reply ckanext/switzerland/helpers/backend_helpers.py:218]: That is fair, thank you for looking into this again :)

**Comment by stefina** [comment]
Great this is done, approved :)
